src/Riok.Mapperly/SyntaxProvider/VersionedSyntaxProvider.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Riok.Mapperly.SyntaxProvider;
+
+internal static class VersionedSyntaxProvider
```

Comment:
I'd remove this class and simply name both roslyn version depended classes the same (`SyntaxProvider`). Since only one should be included at a time during compilation this should work.

Reply:
What should I name the files, they can't both be `SyntaxProvider`. Does conditional compilation in the csproj prevent this issue?

Reply:
I'd move them to `Riok.Mapperly` and name the files `SyntaxProvider.Roslyn4.0.cs` and `SyntaxProvider.Roslyn4.4.cs` and the class `SyntaxProvider`.